### ISSUE TITLE
Font Kerning

### ISSUE THREAD

**Issue by joafalves** [mapbox/tiny-sdf#6]
Hello :)

Do you have any plans to support font kerning on this?

What are your thoughts on a possible implementation?

Thanks.

**Comment by mourner** [comment]
I have no idea how to implement this unfortunately. I made this library for CJK glyphs use case, where all characters are the same size. But I'm welcome to any ideas.

**Comment by joafalves** [comment]
I've recently stumbled on this: https://opentype.js.org
It has all the features we needed to adjust the font rendering correctly 👍

**Comment by joafalves** [comment]
We are using it on all-in-one platform though (https://github.com/scarlettgamestudio/scarlett-editor) you might need to adjust for a "tiny" integration :)

**Comment by mourner** [comment]
@monfera good point — probably needs clarification. Although technically nothing prevents drawing TinySDF for whole labels if a use case doesn't require too many. Also, the primary use case for it currently is rendering Chinese/Japanese/Korean glyphs in Mapbox GL JS — so not necessarily monospaced, if the language uses relatively same-size glyphs and doesn't depend on kerning.

**Comment by controversial** [comment]
You can calculate the kerning for a pair of characters on a canvas by comparing the measured width of the combined characters to the measured width of the characters individually:

```js
const unkernedWidth = tinySdf.ctx.measureText('A').width + tinySdf.ctx.measureText('V').width;
const kernedWidth = tinySdf.ctx.measureText('AV').width;
const kerning = kernedWidth - unkernedWidth; // a negative value indicates you should adjust the SDFs closer together by that much
```
